### PR TITLE
Better docs.

### PR DESCRIPTION
**Pull request by taranjeet**
These are just draft of the docs, Will be updating in this PR only.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+## Glossary
```
Comment (deshraj) [inline docs/glossary.md]: I have updated the glossary part at https://github.com/Cloud-CV/EvalAI/wiki/Glossary. Please copy the markdown code.

**Comment by deshraj** [comment]
@taranjeet please let me know whenever this is ready for review. Let's get the first version merged and then we can iterate over it later on.

**Comment by taranjeet** [comment]
@deshraj : sure.

**Comment by deshraj** [comment]
@taranjeet can we please get this merged ASAP?

**Comment by taranjeet** [comment]
@deshraj : you ca review docs in this PR. Docs which you feel are complete, can be merged right now in to master. I will create a separate PR for those. 

For the rest (docs which are incomplete), please review them, so that I can further update them.

File: docs/architecture_decisions.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+## Architecture Decisions
```
Comment (deshraj) [inline docs/architecture_decisions.md]: Can we replace the content with https://github.com/Cloud-CV/EvalAI/wiki/Architectural-Decisions? I have fixed some of the grammatical mistakes in this.

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+## Architecture
```
Comment (deshraj) [inline docs/architecture.md]: Can we replace it with https://github.com/Cloud-CV/EvalAI/wiki/Architecture? Minor changes done.

File: docs/directory_structure.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+## Directory Structure
```
Comment (deshraj) [inline docs/directory_structure.md]: Can we replace it with https://github.com/Cloud-CV/EvalAI/wiki/Directory-Structure? Minor styling and grammatical fixes.

**Comment by taranjeet** [comment]
@deshraj : I have updated this PR as per your changes. Lets merge this PR first and then I will add more docs in another PR.

**Comment by deshraj** [comment]
Looks good. Merging this one.

**Pull request merged**